src/app/[locale]/(private)/app-sidebar.tsx

```diff
@@ -0,0 +1,164 @@
+import {
+  Sidebar,
+  SidebarContent,
+  SidebarGroup,
+  SidebarGroupContent,
+  SidebarHeader,
+  SidebarMenu,
+  SidebarMenuButton,
+  SidebarMenuItem,
+} from "@/components/ui/sidebar"
+import { useTranslations } from 'next-intl';
+import {
+  Gear,
+  House,
+  UserCircle,
+  CircleWavyCheck,
+  Exam,
+  ListNumbers,
+  GraduationCap,
+  Books,
+  ChartBarHorizontal,
+  Notebook,
+  IdentificationBadge,
+  ChatCenteredText,
+} from '@/app/images';
+import { Link, usePathname } from '@/i18n/routing';
+import { Paragraph } from '@/components/typography/paragraph';
+import { TextButton } from '@/components/ui/text-button';
+import dayjs from 'dayjs';
+import React from 'react';
+
+const createMenuGroups = (t: ReturnType<typeof useTranslations>) => ([
+  [
+    {
+      title: t('main'),
+      url: "/",
+      icon: House,
+    },
+  ],
+  [
+    {
+      title: t('profile'),
+      url: "/profile",
+      icon: UserCircle,
+    },
+    {
+      title: t('current-superintendence'),
+      url: "/current-superintendence",
+      icon: CircleWavyCheck,
+    },
+    {
+      title: t('certification-results'),
+      url: "/certification-results",
+      icon: Exam,
+    },
+  ],
+  [
+    {
+      title: t('disciplines'),
+      url: "/disciplines",
+      icon: ListNumbers,
+    },
+    {
+      title: t('term'),
+      url: "/term",
+      icon: GraduationCap,
+    },
+    {
+      title: t('methodological-support'),
+      url: "/methodological-support",
+      icon: Books,
+    },
+    {
+      title: t('poll'),
+      url: "/poll",
+      icon: ChartBarHorizontal,
+    },
+    {
+      title: t('curriculum'),
+      url: "/curriculum",
+      icon: Notebook,
+    },
+    {
+      title: t('contacts'),
+      url: "/contacts",
+      icon: IdentificationBadge,
+    },
+  ],
+  [
+    {
+      title: t('feedback'),
+      url: "/feedback",
+      icon: ChatCenteredText,
+    },
+    {
+      title: t('settings'),
+      url: "/settings",
+      icon: Gear,
+    },
+  ],
+]);
+
+const createSubMenuLinks = (t: ReturnType<typeof useTranslations>) => ([
+  { title: t('user-manual'), url: '/user-manual' },
+  { title: t('faq'), url: '/frequently-asked-questions' },
+  { title: t('about'), url: '/about' },
+  { title: t('documents'), url: '/documents' },
+  { title: t('terms-of-service'), url: '/terms-of-service' },
+]);
+
+export function AppSidebar() {
+  const t = useTranslations('global.menu');
+  const pathname = usePathname();
+  const mainMenuGroups = createMenuGroups(t);
+  const subMenuLinks = createSubMenuLinks(t);
+
+  return (
+    <Sidebar>
+      <SidebarContent>
+        <SidebarHeader className="py-[16px] px-[calc(16px_+_0.5rem)]">
+          <div className="flex max-h-[48px]">
+            <Logo />
+          </div>
+        </SidebarHeader>
+        <SidebarContent className="gap-0">
+          {mainMenuGroups.map((group, index) => (
+            <SidebarGroup className="[&:not(:first-child)]:pt-6 pb-6" key={index}>
```

Comment:
It's not recommended using indexes for keys. Maybe use groupId for example

Reply:
Using indexes is fine as long as elements are not shuffled. Here they are static, so it won't influence on anything.